packages/timeline-state-resolver/src/lib.ts

```diff
@@ -217,3 +220,39 @@ export function endTrace(trace: Trace): FinishedTrace {
 export function deferAsync(fn: () => Promise<void>, catcher: (e: unknown) => void): void {
 	fn().catch(catcher)
 }
+
+/**
+ * Set a value on an object from a .-delimited path
+ * @param obj The base object
+ * @param path Path of the value to set
+ * @param val The value to set
+ */
+const set = (obj: Record<string, any>, path: string, val: any) => {
+	const p = path.split('.')
+	p.slice(0, -1).reduce((a, b) => (a[b] ? a[b] : (a[b] = {})), obj)[p.slice(-1)[0]] = val
+}
+export function fillStateFromDatastore(state: TimelineState, datastore: Datastore) {
```

Comment:
Personal opinion: This should probably moved to a separate file, since it has dependencies to `conductor.ts` it creates a circular dependency between `conductor.ts` and `lib.ts`.